src/Microsoft.Diagnostics.NETCore.Client/ReversedServer/ReversedDiagnosticsServer.cs

```diff
@@ -18,23 +18,30 @@ namespace Microsoft.Diagnostics.NETCore.Client
     /// Establishes server endpoint for runtime instances to connect when
     /// configured to provide diagnostic endpoints in reverse mode.
     /// </summary>
-    internal sealed class ReversedDiagnosticsServer : IDisposable
```

Comment:
Not that I'm strongly advocating for this, but do we need a sync and async version of this api surface? (Including Accept)

Reply:
For completeness, yes, we probably should have the following non-async methods:
- `IpcEndpointInfo ReversedDiagnosticsServer.Accept(TimeSpan)`
- `void ReverseDiagnosticsServer.WaitForConnection(TimeSpan)`
- `void IpcEndpoint.WaitForConnection(TimeSpan)`
- `void DiagnosticsClient.WaitForConnection(TimeSpan)`

And for full parity, we should add the following async methods:
- `Task<Stream> IpcEndpoint.ConnectAsync(CancellationToken)`

However, I'd like to do that in a separate change in order to not make this one very complicated. I think it would involve refactoring the server a bit so that the waiting implementations are not copied again. Also, a refactor of the tests so that way we can write the same test for both non-async and async methods.

We can also consider adding support for async operations in the DiagnosticsClient class, but that would require a lot of changes to the IpcClient, IpcMessage, etc in order to allow fully asynchronous code. If we were to pursue this, I would argue that it should be done as yet another separate change in order to keep the complexity down.